commonci/github.go

```diff
@@ -249,13 +249,18 @@ func (g *GithubRequestHandler) AddPRComment(body *string, owner, repo string, pr
 	return nil
 }
 
-// AddOrEditPRComment posts or edits a comment to a PR.
+// AddEditOrDeletePRComment posts or edits a comment to a PR.
 // If signature is empty, it's equivalent to AddPRComment.
+// Signature is a string that can be found in the comment in order to find and edit it.
 // Otherwise, all comments are searched first, and if any matches the
 // signature, then the comment is edited. If none matches, then a new comment
 // is posted.
-func (g *GithubRequestHandler) AddOrEditPRComment(signature string, body *string, owner, repo string, prNumber int) error {
+// If body is nil, then it indicates delete.
+func (g *GithubRequestHandler) AddEditOrDeletePRComment(signature string, body *string, owner, repo string, prNumber int) error {
 	if signature == "" {
+		if body == nil {
+			return fmt.Errorf("PR comment body unspecified")
+		}
 		return g.AddPRComment(body, owner, repo, prNumber)
```

Comment:
This is not related to the rest of the changes. Just adding additional functionality to the function.